improver/cli/nbhood_land_and_sea.py

```diff
@@ -205,8 +205,10 @@ def process(
     # Section for combining land and sea points following land and sea points
     # being neighbourhood processed individually.
     if sea_only.data.max() > 0.0 and land_only.data.max() > 0.0:
-        # Recombine cubes to be a single output.
+        # Recombine cubes to be a single output, and burn back in the
+        # missing data mask from the original cube.
         combined_data = result_land.data.filled(0) + result_sea.data.filled(0)
+        combined_data = np.ma.masked_array(combined_data, cube.data.mask, copy=False)
```

Comment:
It would be good to add an acceptance test for this to show the behaviour. 

Something like as shown below, based on the existing `test_basic` would show the effect. I've pushed up some example acceptance test data to https://github.com/metoppv/improver_test_data/tree/improver_1937_mask_values. 
```
@pytest.mark.parametrize(
    "kgo_name, shape",
    (("kgo_with_mask.nc", "square"), ("kgo_circular_with_mask.nc", "circular")),
)
def test_with_mask(tmp_path, kgo_name, shape):
    """Test basic land-sea without topographic bands"""
    kgo_dir = acc.kgo_root() / "nbhood-land-and-sea/no_topographic_bands"
    kgo_path = kgo_dir / kgo_name
    input_path = kgo_dir / "input_with_mask.nc"
    mask_path = kgo_dir / "ukvx_landmask.nc"
    output_path = tmp_path / "output.nc"
    args = [
        input_path,
        mask_path,
        "--neighbourhood-shape",
        shape,
        "--radii",
        "20000",
        "--output",
        output_path,
    ]
    run_cli(args)
    acc.compare(output_path, kgo_path)
```

Example input with mask:
![nbhood_land_and_sea_with_mask_input](https://github.com/metoppv/improver/assets/26933341/a0508af9-5732-4a68-bc2d-b2c88304d02a)

Example output with mask:
![nbhood_land_and_sea_with_mask_kgo](https://github.com/metoppv/improver/assets/26933341/01b262ae-7c91-4333-8f32-38c8bd46929c)